library/src/main/java/dev/chrisbanes/insetter/Insetter.java

```diff
@@ -31,13 +31,144 @@
+     * @see Insetter#setOnApplyInsetsListener(View)
+     */
+    @NonNull
+    public Builder setOnApplyInsetsListener(OnApplyInsetsListener onApplyInsetsListener) {
+      this.onApplyInsetsListener = onApplyInsetsListener;
+      return this;
+    }
+
+    /**
+     * @param dimensions Enum set defining padding handling of system window insets.
+     *                   Ignored if {@link Insetter#onApplyInsetsListener } is set.
+     * @see Insetter#applyInsetsToView(View, WindowInsetsCompat, ViewState)
+     */
+    @NonNull
+    public Builder applySystemWindowInsetsToPadding(@Nullable EnumSet<InsetDimension> dimensions) {
+      if (dimensions != null && !dimensions.isEmpty()) {
+        paddingSystemWindowInsets = dimensions;
+      }
+
+      return this;
+    }
+
+    /**
+     * @param dimensions Enum set defining margin handling of system window insets.
+     *                   Ignored if {@link Insetter#onApplyInsetsListener } is set.
+     * @see Insetter#applyInsetsToView(View, WindowInsetsCompat, ViewState)
+     */
+    @NonNull
+    public Builder applySystemWindowInsetsToMargin(@Nullable EnumSet<InsetDimension> dimensions) {
+      if (dimensions != null && !dimensions.isEmpty()) {
+        marginSystemWindowInsets = dimensions;
+      }
+
+      return this;
+    }
+
+    /**
+     * @param dimensions Enum set defining padding handling of system gesture insets.
+     *                   Ignored if {@link Insetter#onApplyInsetsListener } is set.
+     * @see Insetter#applyInsetsToView(View, WindowInsetsCompat, ViewState)
+     */
+    @NonNull
+    public Builder applySystemGestureInsetsToPadding(@Nullable EnumSet<InsetDimension> dimensions) {
+      if (dimensions != null && !dimensions.isEmpty()) {
+        paddingSystemGestureInsets = dimensions;
+      }
+
+      return this;
+    }
+
+    /**
+     * @param dimensions Enum set defining margin handling of system gesture insets.
+     *                   Ignored if {@link Insetter#onApplyInsetsListener } is set.
+     * @see Insetter#applyInsetsToView(View, WindowInsetsCompat, ViewState)
+     */
+    @NonNull
+    public Builder applySystemGestureInsetsToMargin(@Nullable EnumSet<InsetDimension> dimensions) {
+      if (dimensions != null && !dimensions.isEmpty()) {
+        marginSystemGestureInsets = dimensions;
+      }
+
+      return this;
+    }
+
+    /**
+     * @param consumeSystemWindowInsets true if the system window insets should be consumed,
+     *                                  false if not
+     * @see Insetter#setOnApplyInsetsListener(View)
+     */
+    @NonNull
+    public Builder consumeSystemWindowInsets(boolean consumeSystemWindowInsets) {
+      this.consumeSystemWindowInsets = consumeSystemWindowInsets;
+      return this;
+    }
+
+    /**
+     * @param view the {@link View} on which {@link WindowInsetsCompat} should be applied
+     */
+    public void applyToView(@NonNull View view) {
+      build().setOnApplyInsetsListener(view);
+    }
```

Comment:
Not sure if this needs to be part of the builder

Reply:
I like this. I'd make the doc a bit clearer though. Something like:

> Builds the `Insetter` instance and sets it as a `OnApplyWindowInsetsListener` on [view].

Might also be worth returning the built `Insetter` from the function too.